src/ArrayHelper.php

```diff
@@ -428,20 +432,44 @@ public static function setValueByPath(array &$array, $path, $value, string $deli
 
     /**
      * @param array|float|int|string $path The path of where do you want to write a value to `$array`.
-     * The path can be described by a string when each key should be separated by delimiter.
+     * The path can be described by a string when each key should be separated by delimiter. If a path item contains
+     * delimiter, it can be escaped with "\" (backslash) or a custom delimiter can be used.
      * You can also describe the path as an array of keys.
-     * @param string $delimiter A separator, used to parse string $key for embedded object property retrieving. Defaults
+     * @param string $delimiter A separator, used to parse string key for embedded object property retrieving. Defaults
      * to "." (dot).
+     * @param bool $unescapeDelimiter Whether to unescape delimiter in the items of final array (in case pf using string
+     * as an input) Defaults to `true`, meaning "\" (backslashes) are removed - "\." will become "." for "." delimiter.
      *
      * @psalm-param ArrayPath $path
      *
      * @return array|float|int|string
      * @psalm-return ArrayKey
      */
-    private static function parsePath($path, string $delimiter)
+    public static function parsePath($path, string $delimiter = '.', bool $unescapeDelimiter = true)
```

Comment:
I'd name it a bit differently and invert logic / description:

```suggestion
    public static function parsePath($path, string $delimiter = '.', bool $escapeDelimiter = false)
```